src/ape_plugins/_cli.py

```diff
@@ -22,38 +23,82 @@ def cli():
     """
 
 
-@cli.command(name="list", short_help="List installed plugins")
+def _display_section(header: str, lines: List[Set[str]]):
+    click.echo(header)
+    for output in lines:
+        if output:
+            formatted_output = _format_output(output)
+            click.echo("  {}".format("\n  ".join(formatted_output)))
+
+
+def _format_output(plugins_list: Set[str]) -> Set:
+    output = set()
+    for i in plugins_list:
+        text = i.replace("ape_", "")
+        output.add(text)
+    return output
+
+
+@cli.command(name="list", short_help="Display plugins")
 @click.option(
     "-a",
     "--all",
     "display_all",
     default=False,
     is_flag=True,
-    help="Display all plugins (including Core)",
+    help="Display all plugins installed and available (including Core)",
 )
 @ape_cli_context()
 def _list(cli_ctx, display_all):
-    plugins = []
+    installed_first_class_plugins = set()
+    installed_second_class_plugins = set()
+    installed_second_class_plugins_no_version = set()
+    installed_third_class_plugins = set()
+
     for name, plugin in plugin_manager.list_name_plugin():
-        version_str = ""
         version = get_package_version(name)
 
         if name in FIRST_CLASS_PLUGINS:
             if not display_all:
-                continue  # NOTE: Skip 1st class plugins unless specified
-            version_str = " (core)"
-
-        elif version:
-            version_str = f" ({version})"
-
-        plugins.append(f"{name}{version_str}")
 
-    if plugins:
-        click.echo("Installed plugins:")
-        click.echo("  " + "\n  ".join(plugins))
-
-    else:
-        cli_ctx.logger.info("No plugins installed")
+                continue  # NOTE: Skip 1st class plugins unless specified
+            installed_first_class_plugins.add(f"{name}")
+
+        elif name in SECOND_CLASS_PLUGINS:
+            installed_second_class_plugins.add(f"{name}     {version}")
+            installed_second_class_plugins_no_version.add(f"{name}")
+
+        elif name not in FIRST_CLASS_PLUGINS or name not in SECOND_CLASS_PLUGINS:
+            installed_third_class_plugins.add(f"{name}      {version})")
+        else:
+            cli_ctx.logger.error(f"{name} is not a plugin.")
+
+    # First Class Plugins
+    if display_all:
+        _display_section("Installed Core Plugins:", [installed_first_class_plugins])
+        click.echo()
+
+    # Second and Third Class Plugins
+    available_second = list(SECOND_CLASS_PLUGINS - installed_second_class_plugins_no_version)
+
+    if installed_second_class_plugins:
+
+        _display_section(
+            "Installed Plugins:", [installed_second_class_plugins, installed_third_class_plugins]
+        )
+    elif not display_all:
+        if SECOND_CLASS_PLUGINS:
+            click.echo("No plugins installed")
+
+    if display_all:
+        click.echo()
+        available_second_output = _format_output(available_second)
+        if available_second_output:
+            click.echo()
```

Comment:
There's an extra `click.echo()` in here some where adding a new line:
```bash
$ ape plugins list
Installed Plugins:
  vyper     0.1.0a4
$ ape plugins list -a
Installed Core Plugins:
  ethereum
  accounts
  networks
  compile
  test
  geth
  pm
  plugins
  console
  run

Installed Plugins:
  vyper     0.1.0a4


Available Plugins:
  hardhat
  ledger
  etherscan
  tokens
  solidity
  trezor
  ens
  infura
$
```
Note extra space after vyper

```bash
$ unset GITHUB_ACCESS_TOKEN 
$ ape plugins list
WARNING: $GITHUB_ACCESS_TOKEN not set, unable to list all plugins
$ ape plugins list -a
WARNING: $GITHUB_ACCESS_TOKEN not set, unable to list all plugins
Installed Core Plugins:
  test
  run
  compile
  console
  ethereum
  accounts
  networks
  pm
  plugins
  geth


$ 
```
Note extra space at the end of listing installed

Reply:
This should be taken care of now!